### ISSUE TITLE
Reporting probably incorrect image orientation

### ISSUE THREAD

**Issue by zdila** [mattiasw/ExifReader#27]
ExifReader returns Orientation of value 6 for picture https://user-images.githubusercontent.com/26544966/30562727-cfe7e2f2-9cbf-11e7-82f6-dbb969e79380.jpg. Exiftool returns "Horizontal (normal)", also as https://www.thexifer.net/. Also exiftran doesn't rotate the picture. Just http://metapicz.com shows two orientations - one from EXIF as "Horizontal (normal)" and other from XMP as "Rotate 90 CW".

Maybe the metadata is simply wrong but ExifReader should return 1 to be compatibile with probably most of the other tools.

**Comment by mattiasw** [comment]
So, you've encountered a problem I was worried about after I had added the XMP support. I didn't take into account when an XMP field has the same name as a field in the Exif data. That now manifests itself in the problem you're seeing: the XMP is overwriting the Exif value.

I guess the solution is to separate the different data areas. That would mean a different output format, maybe something like this:
```
{
	"exif": {
		"Orientation": {
			"value": 1,
			"description": "Horizontal (normal)"
		}
	},
	"xmp": {
		"Orientation": {
			"value": 6,
			"description": "Rotate 90 CW"
		}
	}
}
```

A drawback of this (except the breaking change of the format) is that you can't just say you want the "Orientation" and you get the orientation of the image. You have to specify which "Orientation", Exif or XMP. But maybe that's not a problem?

Another way to go would be something like this:
```
{
	"Orientation": {
		"value": 1,
		"description": "Horizontal (normal)"
	}
	"xmp:Orientation": {
		"value": 6,
		"description": "Rotate 90 CW"
	}
}
```

(Or actually it would probably be "tiff:Orientation" since that's what's in the XMP data in this case.)

I think I'm leaning towards the second structure since that would not break as much of the output format and would also work if there are duplicate tag names in the XMP itself. What do you think?

**Comment by zdila** [comment]
I like that second structure :-).

**Comment by mattiasw** [comment]
Good. :-) I'll try implementing this and we'll see how it goes. Can't say a time estimate though I'm afraid.

**Comment by disarticulate** [comment]
I've routed around this issue in a fork:
https://github.com/disarticulate/ExifReader

My solution was to simply tack on a options:
``` {flat:true} ```

Which gives the default behavior.

flat: false would seperate th tags into xmp, itrc, tiff:
```
export function load(data, options = {flat: true}) {
    let dataView;

    try {
        dataView = new DataView(data);
    } catch (error) {
        console.warn('Warning: A full DataView implementation is not available. If you\'re using Node.js you probably want to do this:\n  1. Install a DataView polyfill, e.g. jdataview: npm install --save jdataview\n  2. Require that at the top of your script: global.DataView = require(\'jdataview\');\nSee an example of this in the ExifReader example directory.');  // eslint-disable-line no-console
        return {};
    }

    return loadView(dataView, options);
}

export function loadView(dataView, options = {flat: true}) {
    let foundMetaData = false;
    let tags = {};

    ImageHeader.check(dataView);
    const {tiffHeaderOffset, iptcDataOffset, xmpDataOffset, xmpFieldLength} = ImageHeader.parseAppMarkers(dataView);

    if (hasExifData(tiffHeaderOffset)) {
        foundMetaData = true;
        const tiff = Tags.read(dataView, tiffHeaderOffset);
        (options.flat)
            ? tags = Object.assign({}, tags, tiff)
            : tags.tiff = tiff;
    }
    if (hasIptcData(iptcDataOffset)) {
        foundMetaData = true;
        const iptc = IptcTags.read(dataView, iptcDataOffset);
        (options.flat)
            ? tags = Object.assign({}, tags, iptc)
            : tags.iptc = iptc;
    }
    if (hasXmpData(xmpDataOffset)) {
        foundMetaData = true;
        const xmp = XmpTags.read(dataView, xmpDataOffset, xmpFieldLength);
        (options.flat)
            ? tags = Object.assign({}, tags, xmp)
            : tags.xmp = xmp;
    }
    if (!foundMetaData) {
        throw new Error('No Exif data');
    }

    return tags;
}
```
This way there's no breaking change for current usage, but you can get the seperate tags.

I'll also note that node testing and browser testing give seperate/disparate results.,

I can't tell if Chrome is adding the EXIF data or something, but my images in the node tests do not show an xmp tags, but in the browser the xmp tags overwrite as described above.

**Comment by mattiasw** [comment]
So, I think @disarticulate had a good solution so I went with that (with some minor tweaks). In pull request #32 I (hopefully) solved the XMP overwriting and the description properties for the XMP orientation and GPS tags (#28). Please have a look.

**Comment by mattiasw** [comment]
Fixed in #32.
